command/plugin_list.go

```diff
@@ -127,3 +138,28 @@ func (c *PluginListCommand) Run(args []string) int {
 		return OutputData(c.UI, res)
 	}
 }
+
+func (c *PluginListCommand) simpleResponse(plugins *api.ListPluginsResponse) []string {
+	var flattenedNames []string
+	namesAdded := make(map[string]bool)
+	for _, names := range plugins.PluginsByType {
+		for _, name := range names {
+			if ok := namesAdded[name]; !ok {
```

Comment:
Out of curiosity, how is the data here duplicated? I was under the impression that plugin names had to be unique.

Reply:
You can actually have multiple plugins with the same name and different versions:
```
➜  vault-builtin-deprecation (d86bb29) ✗ /Users/mpalmi/scm/vault/__debug_bin plugin register --version=1.0.0 --sha256=eb9fbddd25b97b763789c7365543d7bb04ca44ad010f88c2a475b1f5fe6c1f12 auth app-id
Success! Registered plugin: app-id
➜  vault-builtin-deprecation (d86bb29) ✗ /Users/mpalmi/scm/vault/__debug_bin plugin list -detailed
Name                             Type        Version                                          Deprecation Status
----                             ----        -------                                          ------------------
alicloud                         auth        v0.12.0+builtin                                  supported
app-id                           auth        v1.0.0                                           n/a
app-id                           auth        v1.12.0+builtin.vault                            pending removal
```

Since the `plugin list` view doesn't show extra information, it only really makes sense to display the unique names. That said, this code predates plugin versioning, so I'm not entirely certain why it was here... I think it may have been for plugins of the same name with different sha's?

Reply:
I think this was actually for plugins with the same name but different types. e.g. we have both auth and secret engines for a lot of the cloud providers.

Maintaining the same simple table of unique names makes even less sense now that we have versioned plugins too. Doesn't have to be in this PR but perhaps we should consider an update to the simple format.

Reply:
Sounds good! I'll open a Jira for this.